contracts/CurrencyNetwork.sol

```diff
@@ -54,23 +54,23 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
     struct Account {
         // A < B (A is the lower address)
 
-        uint128 creditlineGiven;        //  creditline given by A to B, always positive
-        uint128 creditlineReceived;     //  creditline given by B to A, always positive
+        uint64 creditlineGiven;        //  creditline given by A to B, always positive
+        uint64 creditlineReceived;     //  creditline given by B to A, always positive
 
-        int16 interestRateGiven;      //  interest rate set by A for creditline given by A to B in 0.001% per year
-        int16 interestRateReceived;   //  interest rate set by B for creditline given from B to A in 0.001% per year
+        int16 interestRateGiven;      //  interest rate set by A for creditline given by A to B in 0.01% per year
+        int16 interestRateReceived;   //  interest rate set by B for creditline given from B to A in 0.01% per year
 
         uint16 feesOutstandingA;       //  fees outstanding by A
         uint16 feesOutstandingB;       //  fees outstanding by B
 
         uint32 mtime;                  //  last modification time
 
-        int136 balance;                 //  balance between A and B, balance is >0 if B owes A, negative otherwise. balance(B,A) = - balance(A,B)
+        int72 balance;                 //  balance between A and B, balance is >0 if B owes A, negative otherwise. balance(B,A) = - balance(A,B)
```

Comment:
I am not sure this is optimal as the first 7 variables will be stored in a struct (2x64 + 4x16 + 32 = 224) and the balance will be stored in another struct. Whereas we know the balance and mtime will be modified together so they should be in the same struct for gas cost optimisation. You can add a blank space of 64 bits to "fill" the first struct.
That being said, I am not even sure we do not update both structs every time we use `_storeAccount()`

(https://solidity.readthedocs.io/en/v0.4.21/miscellaneous.html?highlight=struct)

The rest looks fine.

Reply:
I actually had this filler, dont now what happened to it. I will look at it and try to optimize for gas.